### ISSUE TITLE
Campfire Issues

### ISSUE THREAD

**Issue by WenXin20** [CJMinecraft01/DoubleSlabs#100]
`Minecraft v1.15.2`
`Forge v31.2.31`
`Double Slabs v2.10.1`

- The bottom campfire has culling applied to it.
- On separate issue, if there is a double campfire and the bottom campfire is broken, the top campfire switches to the bottom and anything that is cooking on it is deleted (if top or bottom is broken).
![2020-08-07_13 30 38](https://user-images.githubusercontent.com/12000593/89677266-c048db80-d8b2-11ea-9712-f8be007f95df.png)

**Comment by WenXin20** [comment]
Looks like the campfire wasn't added to the culling blacklist in the config automatically after updating.

**Comment by WenXin20** [comment]
Another couple of things I've noticed:
- `Edit`: Hay blocks placed under hay bales doesn't set the `signal_fire` to true
- Double and double vertical Campfires are not waterlogged, which creates a weird outline.
- Vertical Campfires that are manually waterlogged do not become unlit.
- Double/Vertical Campfires still do not emit light.
- Double Campfires do not damage entities, only the vertical one does.
![2020-08-07_18 49 21](https://user-images.githubusercontent.com/12000593/89696722-c5bc1b00-d8de-11ea-99ae-915ca3b79497.png)

**Comment by CJMinecraft01** [comment]
Thanks for thoroughly testing these. A lot of these are oversights and assumptions I made when creating the double slabs initially. For every double slab, the assumption is made that it cannot be waterlogged and hence does not have a waterlogged property. 
In regards to the lighting issue, does it occur when OptiFine is not installed? I do trigger a lighting update when the state is updated so it's strange that it still is working (I worked when I tested in my dev environment). In regards to the double campfires not damaging, does it damage the player when you stand on top? Also, in regards to breaking the double campfires and it returning to a regular campfire when broken, I can fix the tile entity data not being set but when breaking a double slab, it replaces the block with the original slab. As a campfire is not a slab and cannot be placed on the upper part of the block, it replaces it with the lower half of the block one. I'm not too sure the best solution to fix this unless I make a custom campfire block which will work on the top half. I'll keep you posted and again, thanks for reporting this
